tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -36,14 +36,34 @@ public TraceContext(IDatadogTracer tracer)
         public IDatadogTracer Tracer { get; }
 
         /// <summary>
-        /// Gets or sets sampling priority.
+        /// Gets or sets the trace's sampling priority.
         /// </summary>
+        [Obsolete("Use SamplingDecision")]
         public SamplingPriority? SamplingPriority
         {
-            get => _samplingPriority;
+            get => SamplingDecision?.Priority;
             set
             {
-                SetSamplingPriority(value);
+                if (value is null)
+                {
+                    SetSamplingDecision(null, notifyDistributedTracer: true);
+                }
+                else
+                {
+                    SetSamplingDecision(new SamplingDecision((SamplingPriority)value, SamplingMechanism.Unknown), notifyDistributedTracer: true);
+                }
```

Comment:
TODO: remove this setter and always use `SamplingDecision` or `SetSamplingDecision()`